### DISCUSSION TITLE
Dynamic paths requires SSR in AppRouter with output export

### DISCUSSION THREAD

**Discussion by sergiohgz** [vercel/next.js#48727] [Help]
### Summary

Hi all!

I am creating a client app with Next 13 appDir and static export (as described in https://beta.nextjs.org/docs/configuring/static-export). This app will have some dynamic paths (based on params) that won't be known at compile time, only at runtime. This dynamic path don't work when serving the app, returning 404.

The page will contain a tree like:
- /
- /users
- /users/:id

The components tree in appDir will be like:
- app/
  - page.tsx
  - layout.tsx
  - users/
    - page.tsx
    - [id]/
      - page.tsx

The output will create an index.html, users.html but nothing for users/[id] path.
<img width="199" alt="image" src="https://user-images.githubusercontent.com/14012309/233803476-a1e377d7-1070-44f9-ad4c-ac0f6c0e5c07.png">

I don't know if I am doing something wrong or how to achieve this behavior, I followed the docs shown in section https://beta.nextjs.org/docs/configuring/static-export#dynamic-fetching-with-client-components. Any ideas?

Thank you all!

### Additional information

```js
Output:


Route (app)                                Size     First Load JS
┌ ○ /                                      318 B          80.4 kB
├ ○ /favicon.ico                           0 B                0 B
├ ○ /users                                 463 B            85 kB
└ λ /users/[id]                            461 B            85 kB
+ First Load JS shared by all              74.1 kB
  ├ chunks/450-09a1554f31c4ab7e.js         22.1 kB
  ├ chunks/f40788f3-fc14807e8e26a085.js    50.2 kB
  ├ chunks/main-app-109868805a607b87.js    214 B
  └ chunks/webpack-23701e5bacce31e3.js     1.65 kB

Route (pages)                              Size     First Load JS
─ ○ /404                                   179 B            86 kB
+ First Load JS shared by all              85.8 kB
  ├ chunks/main-acbabe2dd0cc49f2.js        84 kB
  ├ chunks/pages/_app-56e0c85c55b72f40.js  192 B
  └ chunks/webpack-23701e5bacce31e3.js     1.65 kB

λ  (Server)  server-side renders at runtime (uses getInitialProps or getServerSideProps)
○  (Static)  automatically rendered as static HTML (uses no initial props)
```

Info
```
Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 22.4.0: Mon Mar  6 20:59:28 PST 2023; root:xnu-8796.101.5~3/RELEASE_ARM64_T6000
    Binaries:
      Node: 18.15.0
      npm: 9.5.0
      Yarn: 1.22.19
      pnpm: 8.1.0
    Relevant packages:
      next: 13.3.0
      eslint-config-next: 13.3.0
      react: 18.2.0
      react-dom: 18.2.0
```

### Example

https://github.com/sergiohgz/next-static-output

**Comment by rossi-jeff** [comment]
I think I might be experiencing the same issue https://github.com/vercel/next.js/discussions/48639

**Comment by m-dierker** [comment]
I am also hitting this error -- exact same circumstances as shown in the first comment.
